packages/legend-query-builder/src/components/QueryLoader.tsx

```diff
@@ -0,0 +1,596 @@
+                                Delete
+                              </MenuContentItemLabel>
+                            </MenuContentItem>
+                          )}
+                          {renameQuery && (
+                            <MenuContentItem
+                              onClick={showEditQueryNameInput(query.name, idx)}
+                              disabled={
+                                query.owner !== queryLoaderState.currentUserId
+                              }
+                            >
+                              <MenuContentItemLabel>
+                                Rename
+                              </MenuContentItemLabel>
+                            </MenuContentItem>
+                          )}
+                        </MenuContent>
+                      }
+                      menuProps={{
+                        anchorOrigin: {
+                          vertical: 'bottom',
+                          horizontal: 'left',
+                        },
+                        transformOrigin: {
+                          vertical: 'top',
+                          horizontal: 'left',
+                        },
+                        elevation: 7,
+                      }}
+                    >
+                      <MoreVerticalIcon />
+                    </DropdownMenu>
+                    <ChevronRightIcon />
+                  </div>
+                </div>
+              ))}
+              {queryLoaderState.queries.length === 0 && (
+                <BlankPanelContent>
+                  Search by name to find the relevant queries
+                </BlankPanelContent>
+              )}
+              {queryLoaderState.queries.length >=
+                DEFAULT_TYPEAHEAD_SEARCH_LIMIT && (
+                <div className="query-loader__result__section__message">
+                  Too many matching queries found, truncating list to be more
+                  specific
+                </div>
+              )}
+            </>
+          )}
+          <PanelLoadingIndicator
+            isLoading={
+              !queryLoaderState.loadQueriesState.hasCompleted ||
+              renameQueryState.isInProgress
+            }
+          />
+          {!queryLoaderState.loadQueriesState.hasCompleted && (
+            <BlankPanelContent>Loading queries...</BlankPanelContent>
+          )}
+        </div>
+      </>
+    );
+  },
+);
+
+export const QueryLoader = observer(
+  (props: {
+    queryLoaderState: QueryLoaderState;
+    graphManager: AbstractPureGraphManager;
+    loadQuery: (selectedQuery: LightQuery) => void;
+    renameQuery?:
+      | ((
+          selectedQuery: LightQuery,
+          updatedQueryName: string,
+          renameQueryState: ActionState,
+        ) => Promise<void>)
+      | undefined;
+    modalTitle?: string | undefined;
+    moreOptions?:
+      | {
+          loadAsDialog?: boolean;
+          isDeleteSupported?: boolean;
```

Comment:
splitting props about a similar mechanism (i.e. make it a modal) like this is usually bad design